src/libsyntax/parse/parser.rs

```diff
@@ -3095,21 +3103,23 @@ impl<'a> Parser<'a> {
             // RHS must be parsed with more associativity than the dots.
             let next_prec = AssocOp::from_token(&tok).unwrap().precedence() + 1;
             Some(self.parse_assoc_expr_with(next_prec,
-                                                 LhsExpr::NotYetParsed)
-            .map(|x|{
-                hi = x.span.hi;
-                x
-            })?)
+                                            LhsExpr::NotYetParsed)
+                .map(|x|{
+                    hi = x.span.hi;
+                    x
+                })?)
```

Comment:
Just for the record, I think this is _much_ harder to read than the old version with `try!`. You have to parse the entire expression **backwards** from `?` to see what it applies to.